tectonic-stats-extender/Makefile

```diff
@@ -1,4 +1,85 @@
-.PHONY: vendor
+.PHONY: all build container clean test vendor
+
+# Architectures supported: amd64, arm, arm64 and ppc64le
+ARCH ?= amd64
+BUILD_IMAGE ?= golang:1.9-alpine
+BIN = tectonic-stats-extender
+TEST_BIN = tectonic-stats-test
+PKG = github.com/coreos/container-images/tectonic-stats-extender
+REGISTRY ?= quay.io/coreos
+IMAGE = $(REGISTRY)/$(BIN)
+TEST_IMAGE = $(REGISTRY)/$(TEST_BIN)
+VERSION :=
+TAG := $(shell git describe --abbrev=0 --tags HEAD 2>/dev/null)
+COMMIT := $(shell git rev-parse HEAD)
+ifeq ($(VERSION),)
+    ifeq ($(TAG),)
+        VERSION := unknown-$(COMMIT)
+    else
+        ifeq ($(COMMIT), $(shell git rev-list -n1 $(TAG)))
+            VERSION := $(TAG)
+        else
+            VERSION := $(TAG)-$(COMMIT)
+        endif
+    endif
+endif
+DIRTY := $(shell test -z "$$(git diff --shortstat 2>/dev/null)" || echo -dirty)
+VERSION := $(VERSION)$(DIRTY)
+
 vendor: glide.yaml
-	@glide up -v
+	@glide install -v
 	@glide-vc --use-lock-file --no-tests --only-code
+
+build: bin/$(ARCH)/$(BIN)
+test: bin/$(ARCH)/$(TEST_BIN)
+
+bin/$(ARCH)/$(BIN): build-dirs
```

Comment:
Like `bin/$(ARCH)/$(TEST_BIN):` specifying `test/integration_test.go`, it might be nice to specify all remaining go files here, so GNU Make can rebuild only on go file changes.

Reply:
I'll add this for completeness, however note that this target depends on a phony target so this will ALWAYS be re-built, no-matter if the source files are specified and have not changed